bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```diff
@@ -0,0 +1,321 @@
+import org.openhab.core.types.TimeSeries;
+import org.openhab.core.types.TimeSeries.Policy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ForecastSolarObject} holds complete data for forecast
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ForecastSolarObject implements SolarForecast {
+    private final Logger logger = LoggerFactory.getLogger(ForecastSolarObject.class);
+    private final TreeMap<ZonedDateTime, Double> wattHourMap = new TreeMap<ZonedDateTime, Double>();
+    private final TreeMap<ZonedDateTime, Double> wattMap = new TreeMap<ZonedDateTime, Double>();
+    private final DateTimeFormatter dateInputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    private ZoneId zone = ZoneId.systemDefault();
+    private Optional<String> rawData = Optional.empty();
+    private Instant expirationDateTime;
+
+    public ForecastSolarObject() {
+        expirationDateTime = Instant.now();
+    }
+
+    public ForecastSolarObject(String content, Instant expirationDate) {
+        expirationDateTime = expirationDate;
+        if (!content.isEmpty()) {
+            rawData = Optional.of(content);
+            try {
+                JSONObject contentJson = new JSONObject(content);
+                JSONObject resultJson = contentJson.getJSONObject("result");
+                JSONObject wattHourJson = resultJson.getJSONObject("watt_hours");
+                JSONObject wattJson = resultJson.getJSONObject("watts");
+                String zoneStr = contentJson.getJSONObject("message").getJSONObject("info").getString("timezone");
+                zone = ZoneId.of(zoneStr);
+                Iterator<String> iter = wattHourJson.keys();
+                // put all values of the current day into sorted tree map
+                while (iter.hasNext()) {
+                    String dateStr = iter.next();
+                    // convert date time into machine readable format
+                    try {
+                        ZonedDateTime zdt = LocalDateTime.parse(dateStr, dateInputFormatter).atZone(zone);
+                        wattHourMap.put(zdt, wattHourJson.getDouble(dateStr));
+                        wattMap.put(zdt, wattJson.getDouble(dateStr));
+                    } catch (DateTimeParseException dtpe) {
+                        logger.warn("Exception parsing time {} Reason: {}", dateStr, dtpe.getMessage());
+                        return;
+                    }
+                }
+            } catch (JSONException je) {
+                logger.debug("Error parsing JSON response {} - {}", content, je.getMessage());
+            }
+        }
+    }
+
+    public boolean isValid() {
+        return !(wattHourMap.isEmpty() || wattMap.isEmpty());
+    }
+
+    public boolean isExpired() {
+        return expirationDateTime.isBefore(Instant.now());
+    }
+
+    public double getActualEnergyValue(ZonedDateTime queryDateTime) {
```

Comment:
Consider renaming for clarity since this is not a simple **get** method, but rather a calculation based on the provided parameter and object state.

Reply:
I wouldn't change the notion. It's not a calculation based on various parameters with different results. 

Forecast data in json contains values e.g. each hour. `getActual*` is just delivering values if you connect the datapoints of json with lines to provide intermediate values e.g. each knute.